sdk/monitor/OpenTelemetry.Exporter.AzureMonitor/tests/OpenTelemetry.Exporter.AzureMonitor.UnitTest/ConnectionString/ConnectionStringParserTests.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+using NUnit.Framework;
+
+namespace OpenTelemetry.Exporter.AzureMonitor.ConnectionString
+{
+    public class ConnectionStringParserTests
+    {
+        [Test]
+        public void TestDefaultEndpoints()
+        {
+            RunTest(
+                connectionString: "InstrumentationKey=00000000-0000-0000-0000-000000000000",
+                expectedIngestionEndpoint: Constants.DefaultIngestionEndpoint,
+                expectedInstrumentationKey: "00000000-0000-0000-0000-000000000000");
+        }
+
+        [Test]
+        public void TestEndpointSuffix()
+        {
+            RunTest(
+                connectionString: "InstrumentationKey=00000000-0000-0000-0000-000000000000;EndpointSuffix=ai.contoso.com",
+                expectedIngestionEndpoint: "https://dc.ai.contoso.com/",
+                expectedInstrumentationKey: "00000000-0000-0000-0000-000000000000");
+        }
+
+        [Test]
+        public void TestEndpointSuffix_WithExplicitOverride()
+        {
+            RunTest(
+                connectionString: "InstrumentationKey=00000000-0000-0000-0000-000000000000;EndpointSuffix=ai.contoso.com;IngestionEndpoint=https://custom.profiler.contoso.com:444/",
```

Comment:
can we get one more test case for instrumentation key not set `"InstrumentationKey=;EndpointSuffix=ai.contoso.com;IngestionEndpoint=https://custom.profiler.contoso.com:444/"`. For this case we may need to log an event stating Ikey is not specified.

Reply:
Yes! i'll work on that right now